src/SonarQube.Client/Api/SonarQubeService.cs

```diff
@@ -252,5 +253,32 @@ public Uri GetProjectDashboardUrl(string projectKey)
                     request.EventsSince = eventsSince;
                 },
                 token);
+
+        #region IDisposable Support
+        private bool disposedValue = false; // To detect redundant calls
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!disposedValue)
+            {
+                if (disposing)
+                {
+                    IsConnected = false;
+                    messageHandler.Dispose();
+                }
+
+                disposedValue = true;
+            }
+        }
+
+        // This code added to correctly implement the disposable pattern.
+        public void Dispose()
+        {
+            // Do not change this code. Put cleanup code in Dispose(bool disposing) above.
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+        #endregion
```

Comment:
[minor] I usually like to put a comment close to the `endregion` to state what this is closing (e.g. `#endregion // IDisposable Support`).

Also I don't think we have regions globally in the code base of SLVS so I would remove it.